src/modules/verifiable-credentials/verifiable-credentials-base.service.ts

```diff
@@ -93,12 +101,43 @@ export abstract class VerifiableCredentialsServiceBase {
     proof_options: string
   ): Promise<string>;
 
-  constructor(protected readonly _signerService: SignerService) {}
+  constructor(
+    protected readonly _signerService: SignerService,
+    private readonly _storage: VCStorageClient
+  ) {}
+
+  /**
+   * @description The type of the exchange. Only vc-api exchanges currently supported.
+   * @returns credentials query with matching verifiable presentations
+   */
+  async initiateExchange({ type, url }: ExchangeInvitation) {
+    if (type !== VC_API_EXCHANGE) {
+      throw new Error('Only VC-API exchange is supported');
+    }
+
+    const { data: vpRequest } = await axios.post<VpRequest>(url);
+    const presentationDefinitions = vpRequest.query.find(
+      (q) => q.type === VpRequestQueryType.presentationDefinition
+    )?.credentialQuery as PresentationDefinition[];
+
+    const vc: VerifiableCredential<RoleCredentialSubject>[] = (
+      await Promise.all(
+        presentationDefinitions.map((def) =>
+          this._storage.getCredentialsByPresentationDefinition(def)
+        )
+      )
+    )
+      .reduce((prev, curr) => prev.concat(curr), [])
+      .filter(isRoleCredential);
+    const vp = await this.createVerifiablePresentation(vc);
+    return { vp, vpRequest };
```

Comment:
Instead of returning the VCs here, we can return the `SelectResult` and `PresentationDefinition` for Switchboard to display to user for selection and confirmation
```suggestion
    return await Promise.all(
      presentationDefinitions.map(async (presentationDefinition) => {
        const selectResults = await this._storage.getCredentialsByPresentationDefinition(presentationDefinition);
        return {
          presentationDefinition,
          selectResults
        }
      })
    )
```